Allow preservation of numeric strings for Excel

ripvannwinkler opened the pull request:
When exporting a file strictly for use in Excel, it is sometimes desirable to use a special prefix on numeric strings to preserve leading zeros (e.g. for postal codes '00501'). This pull request supports that feature as a configuration parameter.

JoshClose commented:
Thanks for the pull request.

I merged this but made some changes. https://github.com/JoshClose/CsvHelper/commit/5e98a0e73213ed876e9c85666bc64a2d1bfeac7d

1. The unit test wasn't doing anything so I changed it to test the format.
2. I renamed the configuration setting to `UseExcelLeadingZerosFormatForNumerics`. Yeah, it's long, but is more clear to me.
3. I removed the regex. Regex is just too slow when doing it millions of times.
4. I moved the code that adds the format into the writer where the quotes are added.

ripvannwinkler commented:
Cool, no problem. Glad to see you find the idea useful. Wasn't sure how to
best handle some of it, but I figured you'd be able to clean it right up!

On Wed, Jan 28, 2015 at 12:58 AM, Josh Close <notifications@github.com>
wrote:

> Thanks for the pull request.
>
> I merged this but made some changes. 5e98a0e
> <https://github.com/JoshClose/CsvHelper/commit/5e98a0e73213ed876e9c85666bc64a2d1bfeac7d>
>
>    1. The unit test wasn't doing anything so I changed it to test the
>    format.
>    2. I renamed the configuration setting to
>    UseExcelLeadingZerosFormatForNumerics. Yeah, it's long, but is more
>    clear to me.
>    3. I removed the regex. Regex is just too slow when doing it millions
>    of times.
>    4. I moved the code that adds the format into the writer where the
>    quotes are added.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/JoshClose/CsvHelper/pull/341#issuecomment-71789805>.
>
